Sources/SwiftProcessing/Core/UIKitElements/Capture.swift

```diff
@@ -0,0 +1,310 @@
+                             completion: self.preparingCompletionHandler)
+            }
+        }))
+        alertVC.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: { _ in completion?(false) }))
+        show(alert: alertVC)
+    }
+    
+    private func show(error: NSError) {
+        let alertVC = UIAlertController(title: "Error", message: error.localizedDescription, preferredStyle: .alert)
+        alertVC.addAction(UIAlertAction(title: "Ok", style: .cancel, handler: nil ))
+        show(alert: alertVC)
+    }
+}
+
+extension Camera: AVCapturePhotoCaptureDelegate {
+    func capturePhoto(completion: ((UIImage) -> Void)?) { captureCompletionBlock = completion }
+}
+
+extension UIImage {
+    
+    convenience init?(sampleBuffer: CMSampleBuffer, orientation: UIImage.Orientation = .upMirrored) {
+        guard let pixelBuffer = CMSampleBufferGetImageBuffer(sampleBuffer) else { return nil }
+        CVPixelBufferLockBaseAddress(pixelBuffer, .readOnly)
+        defer { CVPixelBufferUnlockBaseAddress(pixelBuffer, .readOnly) }
+        let baseAddress = CVPixelBufferGetBaseAddress(pixelBuffer)
+        let width = CVPixelBufferGetWidth(pixelBuffer)
+        let height = CVPixelBufferGetHeight(pixelBuffer)
+        let bytesPerRow = CVPixelBufferGetBytesPerRow(pixelBuffer)
+        let colorSpace = CGColorSpaceCreateDeviceRGB()
+        let bitmapInfo = CGBitmapInfo(rawValue: CGImageAlphaInfo.premultipliedFirst.rawValue | CGBitmapInfo.byteOrder32Little.rawValue)
+        guard let context = CGContext(data: baseAddress, width: width, height: height,
+                                      bitsPerComponent: 8, bytesPerRow: bytesPerRow,
+                                      space: colorSpace, bitmapInfo: bitmapInfo.rawValue) else { return nil }
+        
+        guard let cgImage = context.makeImage() else { return nil }
+        self.init(cgImage: cgImage, scale: 1, orientation: orientation)
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////
+
+extension UIApplication {
+    private class func topViewController(controller: UIViewController? = UIApplication.shared.keyWindow?.rootViewController) -> UIViewController? {
+        if let navigationController = controller as? UINavigationController {
+            return topViewController(controller: navigationController.visibleViewController)
+        }
+        if let tabController = controller as? UITabBarController {
+            if let selected = tabController.selectedViewController {
+                return topViewController(controller: selected)
+            }
+        }
+        if let presented = controller?.presentedViewController {
+            return topViewController(controller: presented)
+        }
+        return controller
+    }
+    
+    class var topViewController: UIViewController? { return topViewController() }
+}
+
+extension Sketch{
+    
+    open func createCamera(position: String) -> Camera{
```

Comment:
add the possible positions as constants to the main Sketch class. That way createCamera could be called like so

createCamera(FRONT)
createCamera(BACK)